src/lib/homestore/heap_chunk_selector.cpp

```diff
@@ -0,0 +1,38 @@
+#include "heap_chunk_selector.h"
+
+namespace homeobject {
+
+void HeapChunkSelector::add_chunk(csharedChunk& chunk) {
+    uint32_t pdevId = VChunk(chunk).get_pdev_id();
+    auto it = _pdev_heap_map.find(pdevId);
+    if (it == _pdev_heap_map.end()) it = _pdev_heap_map.try_emplace(pdevId, std::make_shared<std::pair<std::mutex, VChunkHeap>>()).first;
+    std::unique_lock<std::mutex> l(it->second->first);
+    it->second->second.emplace(chunk);
+}
+
+csharedChunk HeapChunkSelector::select_chunk(homestore::blk_count_t count, const homestore::blk_alloc_hints& hint) {
+    auto it = _pdev_heap_map.find(hint.dev_id_hint);
+    if (it == _pdev_heap_map.end()) return nullptr;
+    std::unique_lock<std::mutex> l(it->second->first);
```

Comment:
`scoped_lock` or `lock_guard` if not intending to move it.

Reply:
sure, will change